petab_select/candidate_space.py

```diff
@@ -529,6 +602,628 @@ def iterate():
+        sorted list of best models (self.best_models) and construct complements of
+        these models. For all these complements we compute the distance in number of
+        different estimated parameters to all models from history. For each complement
+        we take the minimum of these distances as it's distance to history. Then we
+        choose the complement model with the maximal distance to history.
+        TODO:
+        Next we check if this model is contained in any subspace. If so we choose it.
+        If not we choose the model in a subspace that has least distance to this
+        complement model.
+        """
+
+        most_distance = 0
+        most_distant_indices = []
+
+        for model in self.best_models:
+            model_parameter_ids = list(model.petab_parameters)
+            model_parameters = model.get_parameter_values(
+                parameter_ids=model_parameter_ids
+            )
+
+            model_estimated_parameters = np.array(
+                [p == ESTIMATE for p in model_parameters]
+            ).astype(int)
+            complement_parameters = abs(model_estimated_parameters - 1)
+            # initialize the least distance to the maximal possible value of it
+            complement_least_distance = len(complement_parameters)
+            # get the complement least distance
+            for history_model_id in self.history:
+                history_model = self.history[history_model_id]
+                history_model_parameter_ids = list(
+                    history_model.petab_parameters
+                )
+                history_model_parameters = history_model.get_parameter_values(
+                    parameter_ids=history_model_parameter_ids
+                )
+
+                history_model_estimated_parameters = np.array(
+                    [p == ESTIMATE for p in history_model_parameters]
+                ).astype(int)
+
+                difference = (
+                    history_model_estimated_parameters - complement_parameters
+                )
+                l1_distance = np.abs(difference).sum()
+
+                if l1_distance < complement_least_distance:
+                    complement_least_distance = l1_distance
+
+            # if the complement is further away than the current best one
+            # then save it as the new most distant away one
+            if complement_least_distance > most_distance:
+                most_distance = complement_least_distance
+                most_distant_indices = complement_parameters
+        if len(most_distant_indices) == 0:
+            raise StopIteration("No most_distant model found. Terminating")
+
+        most_distant_parameter_values = [
+            str(index).replace('1', ESTIMATE) for index in most_distant_indices
+        ]
+
+        most_distant_parameters = {
+            parameter_id: index
+            for parameter_id, index in zip(
+                model.parameters, most_distant_parameter_values
```

Comment:
`model.parameters` here, but `model.petab_parameters` in the loop

Reply:
I've replaced in the whole function all petab_parameter_ids with only one from self.best_models[0].petab_parameters